File: Observer/Golang/MaxObserver.go
package main

import (
	"fmt"
	"math"
)

type MaxObserver struct {
	maxCPU int
	maxRAM int
}

func (observer *MaxObserver) update(CPU int, RAM int)  {
	observer.maxCPU = int(math.Max(float64(observer.maxCPU), float64(CPU)))
	observer.maxRAM = int(math.Max(float64(observer.maxRAM), float64(RAM)))

	fmt.Printf("Current Max CPU: %d\nCurrent Max RAM: %d\n", observer.maxCPU, observer.maxRAM)
}

File: Observer/Python 3/MaxObserver.py
from IObserver import IObserver

class MaxObserver(IObserver):
  def __init__(self):
    super().__init__()
    self.maxCPU = 0
    self.maxRAM = 0
  
  def update(self, cpu , ram):
    self.maxCPU = max([self.maxCPU, cpu])
    self.maxRAM = max([self.maxRAM, ram])

    print("Current Max CPU: {}\nCurrent Max RAM: {}".format(self.maxCPU, self.maxRAM))
